src/components/Footer.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { Status } from '../services/Status';
+// eslint-disable-next-line import/no-cycle
+import { TodosContext } from './TodosContext';
+
+export const Footer: React.FC = () => {
+  const {
+    todos,
+    clearForm,
+    setFilter,
+    filter,
+  } = useContext(TodosContext);
+  const [count, setCount] = useState(0);
+
+  const updateCount = () => {
+    const counts = todos.filter(todo => !todo.completed).length;
+
+    setCount(counts);
+  };
+
+  const hasCompleted = todos.some(todo => todo.completed);
+
+  // Performed when you change the task list (Toodos)
+  useEffect(() => {
+    updateCount();
+  }, [todos]);
```

Comment:
Looks like redundant logic. You may have it with out state and useEffect
```suggestion
  const count= todos.filter(todo => !todo.completed).length;
  const hasCompleted = todos.some(todo => todo.completed);

```